src/basic/env-util.c

```diff
@@ -784,3 +785,45 @@ int setenv_systemd_exec_pid(bool update_only) {
 
         return 1;
 }
+
+int env_parse_extension_hierarchies(char ***ret_hierarchies) {
```

Comment:
hmm if you split this out, maybe make this generic, i.e. call this getenv_path_list() or so that takes the env var name to parse as first arg. similar env vars exist a lot after all. (in fact there might be some code elsewhere we could share here?)

Reply:
Done - the few similar usages I've seen have laxer restrictions that we probably don't want for this - ie, no secure_getenv and allow relative paths and so on